common/src/stack/pylib/stack/exception.py

```diff
@@ -96,8 +96,12 @@ def __init__(self, cmd, arg=None):
 
 class ArgNotFound(ArgError):
 
-	def __init__(self, cmd, arg, argtype):
-		super(ArgNotFound, self).__init__(cmd, arg, 'is not a valid %s' % argtype)
+	def __init__(self, cmd, arg, argtype,params = None):
```

Comment:
I'm summoning my inner Chadd: `pep8`.  Put a ` ` after a `,`, unless it's at the end of the line.

Also, an easier to read style, IMO, is to build the string in a conditional and then the rest of the code path can look the same.  _Keep your conditional blocks short; keep your method calls simple.  Don't repeat yourself if you can avoid it._

```
if cond:
    msg = f'not a valid {argtype}'
else:
    param_erros = #etc
    msg = f'not a valid {argtype} with parameters {param_errors}'

super(ArgNotFound, self).__init__(cmd, arg, msg)
```

Obviously, if there's no final `else` you probably need to initialize `msg` to something above the conditional.

This class/exception is very simple, but the more conditionals a chunk of code has (and the longer those conditional blocks are), the harder it is to reason about.  Function calls are, effectively, a branch in the code, so effectively this code *looks like* it branches within its branches.